example/mailto/mailto.cpp

```diff
@@ -0,0 +1,377 @@
+//
+// Copyright (c) 2022 alandefreitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0.
+// https://www.boost.org/LICENSE_1_0.txt
+//
+
+//[example_mailto
+
+/*
+    This example parses a mailto URL into a new
+    view type and prints its components to
+    standard output.
+*/
+
+#include "mailto_grammar.hpp"
+#include <boost/url/url_view.hpp>
+#include <boost/url/url.hpp>
+#include <boost/url/optional.hpp>
+#include <boost/url/grammar/ci_string.hpp>
+#include <boost/url/parse.hpp>
+#include <boost/url/rfc/absolute_uri_rule.hpp>
+#include <boost/url/grammar/digit_chars.hpp>
+#include <boost/url/grammar/parse.hpp>
+#include <boost/url/grammar/vchars.hpp>
+#include <boost/url/grammar/alpha_chars.hpp>
+#include <boost/url/grammar/alnum_chars.hpp>
+#include <boost/url/grammar/token_rule.hpp>
+#include <boost/url/grammar/alnum_chars.hpp>
+#include <iostream>
+#include <algorithm>
+
+namespace urls = boost::urls;
+
+// fwd-declaration for mailto_view
+struct mailto_rule_t;
+
+/// A new url type for mailto URLs
+/**
+    This class represents a URI with the mailto
+    scheme.
+
+    Unlike a urls::url_view, which only represents
+    the general syntax of urls, a mailto_view
+    represents a reference to fields that are
+    relevant to mailto URLs, while ignoring
+    elements of the general syntax
+    that are not relevant to the scheme.
+
+    This allows us to use the general syntax
+    parsers to create a representation that
+    is more appropriate for the specified scheme
+    syntax.
+
+    @par Specification
+    @li <a href="https://www.rfc-editor.org/rfc/rfc6068"
+        >The 'mailto' URI Scheme</a>
+    @li <a href="https://www.rfc-editor.org/rfc/rfc2368"
+        >The mailto URL scheme</a>
+    @li <a href="https://www.rfc-editor.org/rfc/rfc1738"
```

Comment:
maybe you should list the errata URL as well :)

do we need the older RFC?

Reply:
> maybe you should list the errata URL as well :)

Sure.

> do we need the older RFC?